test/e2e/txsimNode.go

```diff
@@ -0,0 +1,71 @@
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/knuu/pkg/knuu"
+)
+
+const (
+	txsimDockerSrcURL = "ghcr.io/celestiaorg/txsim"
+)
+
+func txsimDockerImageName(version string) string {
+	return fmt.Sprintf("%s:%s", txsimDockerSrcURL, version)
+}
+
+type TxSim struct {
+	Name     string
+	Instance *knuu.Instance
+}
+
+func CreateTxSimNode(
+	name, version string,
+	endpoint string,
+	seed int,
+	sequences int,
+	blobRange int,
+	pollTime int,
+	resources Resources,
+	volumePath string,
+) (*TxSim, error) {
+	instance, err := knuu.NewInstance(name)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.SetImage(txsimDockerImageName(version))
+	if err != nil {
+		return nil, err
+	}
+	err = instance.SetMemory(resources.memoryRequest, resources.memoryLimit)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.SetCPU(resources.cpu)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.AddVolumeWithOwner(volumePath, resources.volume, 10001)
```

Comment:
I don't know if this is needed for txsim. `AddVolume()` is required if you want data persistence after a restart.
As txsim produces no local data and you do not intend to restart it, I think you can safely remove this.

Reply:
What about for upgrades. Doesn't that involve restarting the instance?

Reply:
Thank you! Following your suggestion, I removed it, and everything was fine. I've also integrated your suggestion into my other PR [here](https://github.com/celestiaorg/celestia-app/pull/3244).